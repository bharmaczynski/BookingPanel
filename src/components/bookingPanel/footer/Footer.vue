<template>
    <div class="footer">
        <div class="footer__button">
            <button @click="$emit('buttonClicked')">Book</button>
        </div>
        <div class="footer__text">
            <p class="font-size-s">
                Lorem ipsum dolor sit amet, consectetur dipiscing elit, sed do eiusmod
            </p>
        </div>
    </div>
</template>
<style lang="scss" scoped>
.footer {
    width: 100%;
    display: flex;
    flex-direction: column;

    &__button {
        margin: 30px 0 20px;
        padding-bottom: 30px;
        border-bottom: 1px solid $alto;

        button {
            width: 100%;
            border: 0;
            padding: 10px;
            background: $rose;
            color: #fff;
            border-radius: 4px;
            transition: background 0.2s ease;

            &:hover {
                background: $downy;
            }
        }
    }

    &__text {
        padding-left: 30px;
        margin-bottom: 10px;
        text-align: right;
    }
}
</style>
