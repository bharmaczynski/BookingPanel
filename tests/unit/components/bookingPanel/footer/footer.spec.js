import { shallowMount } from '@vue/test-utils';
import Footer from '@/components/bookingPanel/footer/Footer.vue';

describe('Footer', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallowMount(Footer);
        wrapper.vm.$emit = jest.fn();
    });

    it('should match initial snapshot', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it('should emit proper event after clicking button', () => {
        wrapper.find('.footer__button button').trigger('click');

        expect(wrapper.vm.$emit).toHaveBeenCalledWith('buttonClicked');
    });
});
