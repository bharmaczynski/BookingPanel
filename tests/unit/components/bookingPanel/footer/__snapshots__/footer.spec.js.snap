// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match initial snapshot 1`] = `
<div
  class="footer"
>
  <div
    class="footer__button"
  >
    <button>
      Book
    </button>
  </div>
   
  <div
    class="footer__text"
  >
    <p
      class="font-size-s"
    >
      
            Lorem ipsum dolor sit amet, consectetur dipiscing elit, sed do eiusmod
        
    </p>
  </div>
</div>
`;
